### PR TITLE
Issues/fix docker dependencies

### PR DESCRIPTION
**Pull request by agunapal**
## Description

GPU docker was missing nvgpu

Also, noticed that CPU docker image has some missing dependencies: pynvml , wrong version of Pillow

Also, the code to get the TORCH_VERSION gets the latest version and not the version we want.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by agunapal** [comment]
@lxning The original code was not installing common.txt in case of CPU too. Please suggest if the suggested code is a better approach

File: docker/Dockerfile [hunk 1]
```diff
@@ -62,26 +63,29 @@ RUN export USE_CUDA=1
...
+
+WORKDIR "serve"
+
+RUN \
     if echo "$BASE_IMAGE" | grep -q "cuda:"; then \
         # Install CUDA version specific binary when CUDA version is specified as a build arg
         if [ "$CUDA_VERSION" ]; then \
-            python -m pip install --no-cache-dir torch==$TORCH_VER+$CUDA_VERSION torchvision==$TORCH_VISION_VER+$CUDA_VERSION -f https://download.pytorch.org/whl/torch_stable.html; \
-        # Install the binary with the latest CUDA version support
+            python ./ts_scripts/install_dependencies.py --cuda $CUDA_VERSION; \
```
Comment (lxning) [inline docker/Dockerfile:74]: To solve common_gpu.txt installation, you only need ref
https://github.com/pytorch/serve/blob/master/docker/Dockerfile#L75

File: docker/Dockerfile [hunk 2]
```diff
@@ -62,26 +63,29 @@ RUN export USE_CUDA=1
...
+        # Install the binary with the latest CPU image on a CUDA base image
         else \
-            python -m pip install --no-cache-dir torch torchvision; \
+            python ./ts_scripts/install_dependencies.py; \
         fi; \
-        python -m pip install --no-cache-dir -r https://raw.githubusercontent.com/pytorch/serve/master/requirements/common.txt; \
     # Install the CPU binary
     else \
-        python -m pip install --no-cache-dir torch==$TORCH_VER+cpu torchvision==$TORCH_VISION_VER+cpu -f https://download.pytorch.org/whl/torch_stable.html; \
+        python ./ts_scripts/install_dependencies.py; \
```
Comment (lxning) [inline docker/Dockerfile:81]: ditto

File: docker/Dockerfile (hunk 1 above)
Comment (agunapal) [reply docker/Dockerfile:74]: @lxning  There is a problem with the CPU image too. It doesn't have pynvml , torchaudio, torchtext

File: docker/Dockerfile [hunk 3]
```diff
@@ -62,26 +63,29 @@ RUN export USE_CUDA=1
 
 ARG CUDA_VERSION=""
 
-RUN TORCH_VER=$(curl --silent --location https://pypi.org/pypi/torch/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
-    TORCH_VISION_VER=$(curl --silent --location https://pypi.org/pypi/torchvision/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
+RUN git clone https://github.com/pytorch/serve.git
```
[the remark this answers is not in the document]
Comment (msaroufim) [reply docker/Dockerfile:66]: typically can resolve this with `git clone --depth 1` 

or once done with repo just `RUnN rm -rf serve`

**Review by msaroufim — APPROVED** [review]
LGTM just a minor nit on the cloned repo

**Review by msaroufim — APPROVED** [review]
LGTM just a minor nit on the cloned repo

File: docker/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (agunapal) [reply docker/Dockerfile]: added ````-depth 1```

**Review by lxning — APPROVED** [review]

**Pull request merged**